import_data.py

```diff
@@ -60,51 +70,131 @@ def add_missing_keys(counts, count_keys):
     for key in count_keys:
         if key not in counts: counts[key] = 0
 
-with open(sys.argv[1], 'r') as f:
-    lines = list(csv.DictReader(f, delimiter="|"))
-
-    # Some seasonal employees have "Annual Salary" listed as an empty string. For now skip, but log how many
-    # Some records in 20150301.csv have '$-  '
-    parsed = []
-    for ix, l in enumerate(lines):
-        try:
-            l['Annual Salary'] = parse_float(l['Annual Salary'])
-        except ValueError:
-            print("Skipping line {}: Unable to parse '{}' as float".format(ix, l['Annual Salary']))
-            continue
-        l['Income Category'] = income_category(l['Annual Salary'])
-        parsed.append(l)
-
-    # Group by Department
-    by_department = sorted(parsed, key=department)
-    per_department = {k: list(v) for k, v in groupby(by_department, key=department)}
-
-    # For each income group, generate a grouping by gender and demographic
-    department_rollups = {}
-    # We tack on "All Departments" to get rollups for everyone. It's inefficient but it still sub-second execution time
-    for department, values in chain(per_department.items(), [('All Departments', parsed)]):
-        department_rollups[department] = {
-            'ethnicity': [],
-            'gender': [],
+def generate_report(file):
```

Comment:
This is fine in python 3, but `file` is a builtin type in python 2. Use a different name to avoid ambiguity.